herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -1632,7 +1632,14 @@ TableSpaceCheckpoint checkpoint(boolean full, boolean pin, boolean alreadLocked)
                     throw new DataStorageManagerException("actualLogSequenceNumber cannot be null");
                 }
                 // TODO: transactions checkpoint is not atomic
-                actions.addAll(dataStorageManager.writeTransactionsAtCheckpoint(tableSpaceUUID, logSequenceNumber, new ArrayList<>(transactions.values())));
+                Collection<Transaction> currentTransactions = new ArrayList<>(transactions.values());
+                for (Transaction t : currentTransactions) {
+                    LogSequenceNumber txLsn = t.lastSequenceNumber;
```

Comment:
Can there be a case when LogSequenceNumber is null? If yes why and should we log info or error to indicate that?

Reply:
Yes
When you create a transaction it does not have a lsn because we can assign a value only for writes to the log that have been persisted on the CommitLog.
While dealing with transactions we only require that the commit() operation is fsynced to disk, this is particularly needed for BK commit log.
So it is natural for instance that a transaction just created without any dml execution has a null lsn.
So logging this case may lead to a lot of curn into the logs

https://github.com/diennea/herddb/blob/e29eab66acccc9ecb6dffe6b31c8f7fa6fc21e41/herddb-core/src/main/java/herddb/core/TableSpaceManager.java#L326

Reply:
Makes sense.